agentseek/src/main/kotlin/io/github/agentseek/view/input/InputListener.kt

```diff
@@ -8,29 +8,36 @@ import java.awt.event.KeyListener
 import java.awt.event.MouseEvent
 import java.awt.event.MouseListener
 
+
 open class InputListener : KeyListener, MouseListener, InputProvider {
-    private var keyListeners = mutableListOf<(Input.Key) -> Unit>()
+    private var keyListeners = mutableListOf<Pair<(Input.Key) -> Unit, (Input.Key) -> Unit>>()
     private var mouseListeners = mutableListOf<(Input.Mouse) -> Unit>()
+    private var pressedKeySet = mutableSetOf<Int>()
+
+    private var inputKey: Input.Key? = null
 
```

Comment:
Why this needs to be a private field? Can't it be just a `val` inside each method?